### PR TITLE
New onboarding changes android

### PR DESCRIPTION
**Pull request by deeppandya**
Resolves https://github.com/brave/brave-browser/issues/42863

## Submitter Checklist:

- [x] I confirm that no [security/privacy review is needed](https://github.com/brave/brave-browser/wiki/Security-reviews) and no other type of reviews are needed, or that I have [requested](https://github.com/brave/reviews/issues/new/choose) them
- [x] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [x] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [x] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [x] Squashed any review feedback or "fixup" commits before merge, so that history is a record of what happened in the repo, not your PR
- [x] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [ ] Checked the PR locally:
  * `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests` [wiki](https://github.com/brave/brave-browser/wiki/Tests)
  * `npm run presubmit` [wiki](https://github.com/brave/brave-browser/wiki/Presubmit-checks), `npm run gn_check`, `npm run tslint`
- [x] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

## Test Plan:
1. Run brave with clean profile and apply `adb shell "echo ' --enable-features=NewAndroidOnboarding' > /data/local/tmp/brave-test-day-zero-expt"`
2. launch the app
3. observer the onboarding
4. it should behave as show below in the recording. 

VIDEO coming soon

### CODE REVIEW

**Review by samartnik — APPROVED** [review]
lgtm

File: android/java/org/chromium/chrome/browser/firstrun/WelcomeOnboardingActivity.java [hunk 1]
```diff
@@ -231,7 +232,11 @@ private void onClickViews() {
     }
 
     private boolean shouldForceDefaultBrowserPrompt() {
-        return !DayZeroHelper.getDayZeroExptFlag() && !isDefaultBrowser();
+        return isNewOnabordingEnabled() && !isDefaultBrowser();
+    }
+
+    private boolean isNewOnabordingEnabled() {
```
Comment (mherrmann) [inline android/java/org/chromium/chrome/browser/firstrun/WelcomeOnboardingActivity.java:238]:
```suggestion
    private boolean isNewOnboardingEnabled() {
```

File: browser/brave_browser_features.cc [hunk 2]
```diff
@@ -46,6 +46,11 @@ BASE_FEATURE(kBraveDayZeroExperiment,
              "BraveDayZeroExperiment",
              base::FEATURE_DISABLED_BY_DEFAULT);
 
+// Enable new onabording on android
```
Comment (mherrmann) [inline browser/brave_browser_features.cc:49]:
```suggestion
// Enable new onboarding on Android
```

File: android/java/org/chromium/base/BraveFeatureList.java [hunk 3]
```diff
@@ -33,6 +33,7 @@ public abstract class BraveFeatureList {
     public static final String BRAVE_SHOW_STRICT_FINGERPRINTING_MODE =
             "BraveShowStrictFingerprintingMode";
     public static final String BRAVE_DAY_ZERO_EXPERIMENT = "BraveDayZeroExperiment";
+    public static final String BRAVE_NEW_ANDROID_ONBOARDING = "NewAndroidOnboarding";
```
Comment (mherrmann) [inline android/java/org/chromium/base/BraveFeatureList.java:36]: "New onboarding" is very generic. Wouldn't it be clearer to call the feature "AndroidForceDefaultBrowserPrompt"?

**Review by mherrmann — CHANGES_REQUESTED** [review]
Do we also need changes an `about_flags.cc`, as in https://github.com/brave/brave-core/pull/25696/commits/f49e3dfa38b28c861ba62ea611103815932d7caf#diff-cacbc437964cc20aaaf741b80ecbea5cfce9ca9e67725476f73f14cbab2269fb ?

**Comment by deeppandya** [comment]
> Do we also need changes an `about_flags.cc`, as in [f49e3df#diff-cacbc437964cc20aaaf741b80ecbea5cfce9ca9e67725476f73f14cbab2269fb](https://github.com/brave/brave-core/commit/f49e3dfa38b28c861ba62ea611103815932d7caf#diff-cacbc437964cc20aaaf741b80ecbea5cfce9ca9e67725476f73f14cbab2269fb) ?

@mherrmann i don't think we need to expose the flag for users as we can control it through griffin.

File: android/java/org/chromium/base/BraveFeatureList.java (hunk 3 above)
Comment (deeppandya) [reply android/java/org/chromium/base/BraveFeatureList.java:36]: after talking to @kjozwiak  and @timchilds, we are keeping the flag as `NewAndroidOnboarding`

File: browser/brave_browser_features.cc [hunk 4]
```diff
@@ -46,6 +46,11 @@ BASE_FEATURE(kBraveDayZeroExperiment,
              "BraveDayZeroExperiment",
              base::FEATURE_DISABLED_BY_DEFAULT);
 
+// Enable new onboarding on Android
+BASE_FEATURE(kNewAndroidOnboarding,
+             "NewAndroidOnboarding",
+             base::FEATURE_DISABLED_BY_DEFAULT);
+
```
Comment (emerick) [inline browser/brave_browser_features.cc:53]: Shouldn't this only be added on Android platforms?

**Review by emerick — APPROVED** [review]

**Review by DJAndries — APPROVED** [review]
lgtm

**Review by mherrmann — APPROVED** [review]

**Comment by kjozwiak** [comment]
Verification **PASSED** on `Pixel 6` running `Android 15` using the following build(s):

```
Brave | 1.75.110 Chromium: 132.0.6834.46 (Official Build) canary (64-bit)
--- | ---
Revision | ba1e1f0bda21276e78426fcbf9ab0a82d0c3d799
OS | Android 15; Build/AP41.240925.012; 35; REL
```

* installed `1.75.110 Chromium: 132.0.6834.46`
* before launching the browser, connected to `Linux` VM, ran through the following commands via the terminal

```
kjozwiak@kjozwiak-VMUbuntu:~$ adb shell "echo ' --enable-features=NewAndroidOnboarding' > /data/local/tmp/brave-test-day-zero-expt"
kjozwiak@kjozwiak-VMUbuntu:~$ adb shell "cat /data/local/tmp/brave-test-day-zero-expt"
 --enable-features=NewAndroidOnboarding
kjozwiak@kjozwiak-VMUbuntu:~$
```

* ensured that the new `onboarding` is being used that was implemented/tested (`11%` retention increase)
 * https://github.com/brave/brave-core/pull/25696
* ensured that there's no `Notification` system permission modal when Brave first launches
* ensured that the `default browser` system modal appears
* ensured that the P3A/Data onboarding card is still being displayed

image

https://github.com/user-attachments/assets/9c798663-787c-47b2-a0b4-40b6e54b23d1

Also ensured the following so we don't run into issues like https://github.com/brave/brave-browser/issues/42113 & https://github.com/brave/brave-browser/issues/42120:

* ensured that SNTP are being displayed/appearing without any issues
* ensured that `Rewards` icon is still being displayed and the user can opt-in into `rewards`
* ensured that `Brave News` opt-in card is being displayed via NTP
* ensured that `Brave Wallet` appears within the `Hamburger Menu` and is selectable/working without any issues

**`Example`** | **`Example`** | **`Example`**
---------------|-----------------|-----------------
Screenshot_20241218-212434 | Screenshot_20241218-212438 | Screenshot_20241218-212442\

**Pull request merged**